app/services/users/archiver.rb

```diff
@@ -14,6 +14,8 @@ def archive!
         user.archived_email = user.email
         user.archived_at = Time.zone.now
         user.email = "archived-#{user.email}"
+        user.uid = nil
```

Comment:
Even in migration - should we be making the UID nil?
What will happen otherwise is a user will log in with an archived user account, and it will be matched via the UID, and the email, name, etc will be updated to the latest from DfE Identity - and it will actually not really be an archived user - except that `archived_at` will still be populated, and `archived?` will return true.

Reply:
good point, although we are clearing it elsewhere I think worth doing that here anyway let's keep it

Reply:
might be worth just double checking the code to ensure that we never associate anyone with an archived user, I think this helps ensure that, but maybe the last thing would be ecf id, so worth checking the code (possibly without code changes), happy to chat through this